internal/util/helpers_test.go

```diff
@@ -4,18 +4,767 @@
+  "owner": "repoOwner",
+  "name": "repoName"
+}`,
+			expectedError: false,
+		},
+		{
+			name:          "Nil proto message",
+			input:         nil,
+			expectedJson:  `{}`,
+			expectedError: false,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+			jsonResult, err := util.GetJsonFromProto(tt.input)
+			if (err != nil) != tt.expectedError {
+				t.Errorf("GetJsonFromProto() error = %v, expectedError %v", err, tt.expectedError)
+				return
+			}
+			if jsonResult != tt.expectedJson {
+				t.Errorf("GetJsonFromProto() = %v, expected %v", jsonResult, tt.expectedJson)
+			}
+		})
+	}
+}
+
+// TestGetYamlFromProto tests the GetYamlFromProto function
+func TestGetYamlFromProto(t *testing.T) {
+	t.Parallel()
+	tests := []struct {
+		name          string
+		input         proto.Message
+		expectedYaml  string
+		expectedError bool
+	}{
+		{
+			name: "Valid proto message",
+			input: &pb.Repository{
+				Owner: "repoOwner",
+				Name:  "repoName",
+			},
+			expectedYaml: `name: repoName
+owner: repoOwner
+`,
+			expectedError: false,
+		},
+		{
+			name:  "Nil proto message",
+			input: nil,
+			expectedYaml: `{}
+`,
+			expectedError: false,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+			yamlResult, err := util.GetYamlFromProto(tt.input)
+			if (err != nil) != tt.expectedError {
+				t.Errorf("GetYamlFromProto() error = %v, expectedError %v", err, tt.expectedError)
+				return
+			}
+			if yamlResult != tt.expectedYaml {
+				t.Errorf("GetYamlFromProto() = %v, expected %v", yamlResult, tt.expectedYaml)
+			}
+		})
+	}
+}
+
+// TestOpenFileArg tests the OpenFileArg function
+func TestOpenFileArg(t *testing.T) {
+	t.Parallel()
+
+	testCases := []struct {
+		name         string
+		filePath     string
+		dashOpen     io.Reader
+		expectedDesc string
+		expectError  bool
+	}{
+		{
+			name:         "Dash as file path",
+			filePath:     "-",
+			dashOpen:     strings.NewReader("dash input"),
+			expectedDesc: "dash input",
+			expectError:  false,
+		},
+		{
+			name:         "Valid file path",
+			filePath:     "testfile.txt",
+			dashOpen:     nil,
+			expectedDesc: "test content",
+			expectError:  false,
+		},
+		{
+			name:         "Invalid file path",
+			filePath:     "nonexistent.txt",
+			dashOpen:     nil,
+			expectedDesc: "",
+			expectError:  true,
+		},
+	}
+
+	for _, tc := range testCases {
+		tc := tc
+
+		t.Run(tc.name, func(t *testing.T) {
+			t.Parallel()
+
+			// Create a temporary file for testing
+			if tc.filePath == "testfile.txt" {
+				err := os.WriteFile(tc.filePath, []byte(tc.expectedDesc), 0600)
+				assert.NoError(t, err)
+				defer os.Remove(tc.filePath)
+			}
+
+			desc, closer, err := util.OpenFileArg(tc.filePath, tc.dashOpen)
+			if closer != nil {
+				defer closer()
+			}
+
+			if tc.expectError {
+				assert.Error(t, err)
+			} else {
+				assert.NoError(t, err)
+				if tc.filePath == "-" || tc.filePath != "-" {
```

Comment:
Doesn't this `if` cover all cases?

Reply:
The condition if tc.expectError { assert.Error(t, err) } covers only the test cases where an error is expected.

The test cases with no expected error ( there are two)  are covered by the subsequent else block, which checks that no error occurred and verifies the file content.

Reply:
Else condition covers the case:
	    {
			name:         "Dash as file path",
			filePath:     "-",
			dashOpen:     strings.NewReader("dash input"),
			expectedDesc: "dash input",
			expectError:  false,
		}

Reply:
"Alternatively, you could create testfile.txt once outside the t.Run()"

 testfile.txt  is required only in one case, so it is created only for that case.

Reply:
Sorry, the comment here is that this line says:

```suggestion
				path := tc.filePath
				if path == "-" || path != "-" {
```

It should always be true that a string is either equal to a constant, or not equal to the same constant.

Reply:
> "Alternatively, you could create testfile.txt once outside the t.Run()"
> 
> testfile.txt is required only in one case, so it is created only for that case.

(This was from a different comment.)

While it's true that `testfile.txt` is only needed in one case, it doesn't hurt the other cases, and could end up simplifying the setup logic.  On the other hand, I don't feel strongly about it being one place or the other.

Reply:
> Sorry, the comment here is that this line says:
> 
> It should always be true that a string is either equal to a constant, or not equal to the same constant.
I got it, I will fix it.